src/openrct2/ride/Ride.h

```diff
@@ -223,7 +216,7 @@ struct Ride
     std::string custom_name;
     uint16_t default_name_number;
     CoordsXY overall_view;
-    uint16_t vehicles[MAX_VEHICLES_PER_RIDE + 1]; // Points to the first car in the train
+    uint16_t vehicles[OpenRCT2::Limits::MaxTrainsPerRide]; // Points to the first car in the train
```

Comment:
Why did you make this lower?

Even if it is correct, it’s unrelated to this PR.

Reply:
As stated in my comment the last entry was never used and always null. It made for a bad design as you had to have <= instead of < in many places. It now becomes clear that the size is the size and not a bit smaller than the size and we can use all the normal std lib functions on the array and not have to worry about the size not being the size

Reply:
I’d prefer if this was done in a separate PR - it’s not related to this one.

Reply:
Hmm okay it just looks so hacky though.

Reply:
Issue for this has been created will revert changes for now.